dask/dataframe/optimize.py

```diff
@@ -32,3 +36,57 @@ def optimize(dsk, keys, **kwargs):
     )
     dsk, _ = cull(dsk, keys)
     return dsk
+
+
+def optimize_read_parquet_getitem(dsk):
+    # find the keys to optimze
+    from .io.parquet.core import ParquetSubgraph
+
+    read_parquets = [k for k, v in dsk.layers.items() if isinstance(v, ParquetSubgraph)]
+
+    layers_dict = dsk.layers.copy()
+
+    for k in read_parquets:
+        columns = set()
+
+        for dep in dsk.dependents[k]:
+            block = dsk.layers[dep]
+            if not isinstance(block, Blockwise):
+                return dsk
+
+            if len(block.dsk) != 1:
+                # TODO
+                return dsk
+
+            if list(block.dsk.values())[0][0] != operator.getitem:
+                return dsk
```

Comment:
In a future PR it would be nice to try subclassing `Blockwise` into a `Getitem` class.  I think that this would be educational.  I'm curious to see how hard it is to make the current blockwise machinery work with subclasses.